### PR TITLE
Remove create-react-class dependency

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: pkg/docker/containers-view.jsx [hunk 1]
```diff
@@ -372,10 +378,15 @@ var ContainerList = createReactClass({
...
+}
+ContainerList.defaultProps = {
+    client: {},
+    onlyShowRunning: true,
+    filterText: ''
+};
 
-var ImageDetails = createReactClass({
-    render: function () {
+class ImageDetails extends React.Component {
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply pkg/docker/containers-view.jsx]: Ok, I 'll replace all stateless components with the other syntax, but let's do it in separate PR.

**Comment by KKoukiou** [comment]
Looks like all failed tests are not relevant to this change. WIP was removed, it's ready for review.

**Comment by mvollmer** [comment]
I am not super happy about the `bind(this)` situation, in general.  (Not a critic of this PR.)

`createReactClass` would autobind all methods to `this`, but extending React.Component doesn't do that, and we need to compensate for this with explicit binding.

(Autobinding means that `var x = obj.handleClick(); x()` would make it so that `this == obj` inside handleClick even whel called indirectly via `x`.  Without autobinding, it would see `this == undefined`.)

We seem to have two approaches for explicit binding:

- In the constructor, so that code further down can just detach the method
  ```
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  ```

- Each time when a method is detached
  ```
  <input ... onChange={this.changeValue.bind(this, "path")} />
  ```
 
Should we just use both and be happy?  Or should we standardize on one?  I think I prefer the per-use, which is less magical and not a whole lot of code.

There is also this article: https://medium.com/komenco/react-autobinding-2261a1092849

**Comment by martinpitt** [comment]
>  <input ... onChange={this.changeValue.bind(this, "path")} />

TBH I don't like that much. It's not clear whether this would call .bind() for every invocation (which would be inefficient), or just on instantiation (which is still more work than doing it once in the constructor,  if you use the handler more than once).

There's also  a third option:

    onchange={() => this.changeValue("path")}

we use that a lot as well, especially when it's necessary to pass explicit parameters to the handler (like the `"path"` here). IMHO this form is much more explicit and less magical than `.bind(this, extra_arg)`. The bind() is only necessary for the `onevent={this.handler}` form, not for `onevent={(args, ..) => this.handler(moreargs, ..)` one, confusingly.

Given that both forms are rather popular, I'm not sure whether we should enforce using just one, but I don't have a strong opinion.

**Comment by mvollmer** [comment]
>  <input ... onChange={this.changeValue.bind(this, "path")} />
>
> TBH I don't like that much. It's not clear whether this would call .bind() for every invocation (which would be inefficient), or just on instantiation (which is still more work than doing it once in the constructor,  if you use the handler more than once).

This would call bind once every time the property value is evaluated, which is once per call of the `render` method.

> There's also  a third option:
>
>   onchange={() => this.changeValue("path")}

Good point!  I don't think `this.handler.bind(this)` is significantly more expensive than `() => this.handler()` but the latter is more straightforward, so it has my vote as well.

**Pull request merged**